src/python/WMComponent/JobSubmitter/JobSubmitterPoller.py

```diff
@@ -225,8 +230,8 @@ def refreshCache(self):
 
         logging.info("Refreshing priority cache with currently %i jobs", len(self.cachedJobIDs))
 
-        if self.cacheRefreshSize == -1 or len(self.cachedJobIDs) < self.cacheRefreshSize or \
-           self.refreshPollingCount >= self.skipRefreshCount:
+        if self.useReqMgrForCompletionCheck and self.cacheRefreshSize == -1 or \
+           len(self.cachedJobIDs) < self.cacheRefreshSize or self.refreshPollingCount >= self.skipRefreshCount:
             newJobs = self.listJobsAction.execute()
             self.refreshPollingCount = 0
             abortedAndForceCompleteRequests = self.abortedAndForceCompleteWorkflowCache.getData()
```

Comment:
and then remove `self.useReqMgrForCompletionCheck` from the `if` above, otherwise T0 will never refresh the cache. Then add another `if self.useReqMgrForCompletionCheck` here to update `abortedAndForceCompleteRequests`